crates/fuel-core/src/schema/tx/types.rs

```diff
@@ -272,6 +311,47 @@ impl FailureStatus {
     }
 }
 
+#[derive(Debug)]
+pub struct FailureDuringBlockProductionStatus {
+    pub tx_pointer: TxPointer,
+    pub tx_id: Option<TxId>,
+    pub receipts: Option<Vec<fuel_tx::Receipt>>,
+    pub reason: String,
+}
+
+#[Object]
+impl FailureDuringBlockProductionStatus {
+    async fn reason(&self) -> String {
+        self.reason.clone()
+    }
+
+    async fn tx_pointer(&self) -> TxPointer {
+        self.tx_pointer
+    }
+
+    async fn transaction_id(&self) -> Option<TransactionId> {
+        self.tx_id.map(Into::into)
+    }
+
+    #[graphql(complexity = "query_costs().storage_read + child_complexity")]
+    async fn transaction(
+        &self,
+        ctx: &Context<'_>,
+    ) -> Option<async_graphql::Result<Transaction>> {
+        self.tx_id.map(|tx_id| {
+            let query = ctx.read_view()?;
+            let transaction = query.transaction(&tx_id)?;
```

Comment:
Usually pre-confirmed transaction will live in TxPool, not in the database, so I would suggest to go there=D

Reply:
Changed in 38d2733adb4a6ac2e8ef692b1e76054de2437f83

Not really sure here, though. I thought that executor "extracts" transactions from TxPool, so they no longer live there, but I may be not up to speed with the most recent changes to the TxPool and its interactions with Executor.